apps/judicial-system/backend/src/app/modules/case/case.service.ts

```diff
@@ -1319,6 +1320,49 @@ export class CaseService {
     })
   }
 
+  getConnectedIndictmentCases(
+    caseId: string,
+    defendant: Defendant,
+  ): Promise<Case[]> {
+    let whereClause: WhereOptions = {
+      type: CaseType.INDICTMENT,
+      id: { [Op.ne]: caseId },
+      state: [CaseState.RECEIVED],
+    }
+
+    if (defendant.noNationalId) {
+      whereClause = {
+        ...whereClause,
+        [Op.and]: [
+          { '$defendants.national_id$': defendant.nationalId },
+          { '$defendants.name$': defendant.name },
+        ],
+      }
+    } else {
+      const formattedNationalId = formatNationalId(defendant.nationalId)
+      whereClause = {
+        ...whereClause,
+        [Op.or]: [
+          { '$defendants.national_id$': defendant.nationalId },
+          { '$defendants.national_id$': formattedNationalId },
+        ],
+      }
+    }
+
+    return this.caseModel.findAll({
```

Comment:
Please consider the previous comment concerning the number of defendants in a case.

Reply:
Is the idea to support merging cases from different district courts?

Reply:
Do you mean filtering out here whether the court matches before returning? But yes my thinking was I'd handle the filtering higher up so that this would be a more generic function that truly returns all connected indictment cases, if I change it I'd consider renaming it to be more specific, like getConnectedIndictmentCasesByCourt or something like that